File: .tmp_py4013/batch_08_dump/2950_number_of_divisible_substrings.js
// LeetCode 2950 - Number of Divisible Substrings
// https://leetcode.com/problems/number-of-divisible-substrings/

var countDivisibleSubstrings = function(word) {
    const vals = [1,1,2,2,2,3,3,3,4,4,4,5,5,5,6,6,6,7,7,7,8,8,8,9,9,9];
    let ans = 0;
    const n = word.length;
    for (let i = 0; i < n; i++) {
        let sum = 0;
        for (let j = i; j < n; j++) {
            sum += vals[word.charCodeAt(j) - 97];
            if (sum % (j - i + 1) === 0) ans++;
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check13/2950_number_of_divisible_substrings.rs
#![allow(dead_code, unused_variables, unused_mut, unused_assignments, unused_imports)]
struct Solution;
// LeetCode 2950 - Number of Divisible Substrings
// https://leetcode.com/problems/number-of-divisible-substrings/

impl Solution {
    pub fn count_divisible_substrings(word: String) -> i32 {
        let vals = [
            1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 5, 5, 6, 6, 6, 7, 7, 7, 8, 8, 8, 9, 9, 9,
        ];
        let word = word.as_bytes();
        let n = word.len();
        let mut ans = 0;
        for i in 0..n {
            let mut sum = 0;
            for j in i..n {
                sum += vals[(word[j] - b'a') as usize];
                if sum % (j - i + 1) as i32 == 0 {
                    ans += 1;
                }
            }
        }
        ans
    }
}
